pkg/converters/ingress/types/defaults.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The HAProxy Ingress Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+const (
+	defaultSSLOptions = "no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets"
+	// TLS up to 1.2
+	defaultSSLCiphers = "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384"
+	// TLS 1.3
+	defaultSSLCipherSuites = "TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256"
+	// Ref: https://github.com/haproxy/spoa-modsecurity/blob/3c895f3e7dd291dba19d57ba054b277e6fb80ca4/README#L70
+	defaultModsecurityArgs = "unique-id method path query req.ver req.hdrs_bin req.body_size req.body"
+	// Ref: https://github.com/corazawaf/coraza-spoa/blob/179cf897147e165ccc1b8974888d8029737b7af5/doc/config/coraza.cfg#L14
+	DefaultCorazaArgs = "app=hdr(host) id=unique-id src-ip=src src-port=src_port dst-ip=dst dst-port=dst_port method=method path=path query=query version=req.ver headers=req.hdrs body=req.body"
```

Comment:
This can be changed to private.

Reply:
Even if I remove `DefaultCorazaArgs` from instance_test.go, the string still needs to be referenced from `/pkg/converters/ingress/ingress.go`. That was the only place I could find where both the output of `CreateDefaults` and the user-supplied overrides (use-coraza) are both present, which are needed so I can conditionally set the default args.

Since this idea of a "default value that depends on another value" is new, what if we just tell users to set their own custom args in addition to setting `modsecurity-use-coraza: true`? 

I predict many people might set their own `app=hdr(host)` arg anyway to make their Coraza config.yaml easier to write. This would simplify the code too since we wouldn't have these encapsulation issues.

Reply:
> the string still needs to be referenced from /pkg/converters/ingress/ingress.go

Which will be in the same package after moving back to pkg/converters/ingress

> what if we just tell users to set their own custom args in addition to setting modsecurity-use-coraza: true

Sure let's try that approach. Can you describe a bit more what do you have in mind? Just an overview on how would be the user intf - keys and their meaning.

Reply:
I'm thinking the docs we give to users would be essentially:

> In order to use Coraza, you must:
> 1. Set `modsecurity-use-coraza: true` in your values.yaml
> 2. Set `modsecurity-args: "app=hdr(host) id=unique-id src-ip=src src-port=src_port dst-ip=dst dst-port=dst_port method=method path=path query=query version=req.ver headers=req.hdrs body=req.body"`
> 3. Set `extraContainers` to run coraza-spoa ...
> 4. Create a ConfigMap to store the Coraza config.yaml, making sure that you either specify `default_app` or change the `app` arg in `modsecurity-args` to match your application name(s) in config.yaml

 Then all `modsecurity-use-coraza` would do is change the `deny if` conditional and the `message` name.

Reply:
An added bonus of us not hard-coding the default coraza `args` is that we aren't tying ourselves to that `experimental` branch of coraza-spoa, which still seems to be experimenting with the exact args that they expect us to set.

Reply:
I see, there are no magic args, but instead a user copy/paste from doc to a configmap entry. It sounds great, let's make that way.